controllers/packetmachine_controller.go

```diff
@@ -146,85 +170,85 @@ func (r *PacketMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		err = r.reconcileDelete(ctx, machineScope)
 		return ctrl.Result{}, err
 	}
-
 	return r.reconcile(ctx, machineScope)
 }
 
 func (r *PacketMachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
 	log := ctrl.LoggerFrom(ctx)
 
-	clusterToObjectFunc, err := util.ClusterToObjectsMapper(r.Client, &infrav1.PacketMachineList{}, mgr.GetScheme())
+	clusterToPacketMachines, err := util.ClusterToTypedObjectsMapper(mgr.GetClient(), &infrav1.PacketMachineList{}, mgr.GetScheme())
 	if err != nil {
 		return fmt.Errorf("failed to create mapper for Cluster to PacketMachines: %w", err)
 	}
 
-	return ctrl.NewControllerManagedBy(mgr).
-		WithOptions(options).
+	err = ctrl.NewControllerManagedBy(mgr).
 		For(&infrav1.PacketMachine{}).
+		WithOptions(options).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(log, r.WatchFilterValue)).
 		Watches(
-			&source.Kind{Type: &clusterv1.Machine{}},
+			&clusterv1.Machine{},
 			handler.EnqueueRequestsFromMapFunc(util.MachineToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("PacketMachine"))),
 		).
 		Watches(
-			&source.Kind{Type: &infrav1.PacketCluster{}},
-			handler.EnqueueRequestsFromMapFunc(r.PacketClusterToPacketMachines(ctx)),
+			&infrav1.PacketCluster{},
+			handler.EnqueueRequestsFromMapFunc(r.PacketClusterToPacketMachines),
 		).
 		Watches(
-			&source.Kind{Type: &clusterv1.Cluster{}},
-			handler.EnqueueRequestsFromMapFunc(clusterToObjectFunc),
-			builder.WithPredicates(predicates.ClusterUnpausedAndInfrastructureReady(log)),
-		).
-		Complete(r)
+			&clusterv1.Cluster{},
+			handler.EnqueueRequestsFromMapFunc(clusterToPacketMachines),
+			builder.WithPredicates(
+				predicates.ClusterUnpausedAndInfrastructureReady(log),
+			),
+		).Complete(r)
+	if err != nil {
+		return fmt.Errorf("failed setting up with a controller manager: %w", err)
+	}
+	return nil
 }
 
 // PacketClusterToPacketMachines is a handler.ToRequestsFunc to be used to enqeue requests for reconciliation
 // of PacketMachines.
-func (r *PacketMachineReconciler) PacketClusterToPacketMachines(ctx context.Context) handler.MapFunc {
+func (r *PacketMachineReconciler) PacketClusterToPacketMachines(ctx context.Context, o client.Object) []ctrl.Request {
```

Comment:
What's the motivation for this change? I didn't notice it in other CAPI 1.5 upgrades whose FooClustetToFooMachines continue to return handler.MapFunc

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/b2239c69e9ae90200b47d3840bed37ee00592f56/test/infrastructure/docker/internal/controllers/dockermachine_controller.go#L463

Reply:
All I'm seeing here is that the other providers are returning a function that does what's already happening here. What is the benefit of that?